code/modules/mob/living/basic/jungle/venus_human_trap.dm

```diff
@@ -161,76 +177,73 @@
 	lighting_cutoff_green = 35
 	lighting_cutoff_blue = 20
 	faction = list(FACTION_HOSTILE,FACTION_VINES,FACTION_PLANTS)
+	gold_core_spawnable = NO_SPAWN
 	initial_language_holder = /datum/language_holder/venus
 	unique_name = TRUE
-	/// A list of all the plant's vines
-	var/list/vines = list()
-	/// The maximum amount of vines a plant can have at one time
-	var/max_vines = 4
-	/// How far away a plant can attach a vine to something
-	var/vine_grab_distance = 5
+	speed = 1.2
+	melee_attack_cooldown = 1 SECONDS
+	ai_controller = /datum/ai_controller/basic_controller/human_trap
 	/// Whether or not this plant is ghost possessable
 	var/playable_plant = TRUE
 
-/mob/living/simple_animal/hostile/venus_human_trap/Life(seconds_per_tick = SSMOBS_DT, times_fired)
+/mob/living/basic/venus_human_trap/Initialize(mapload)
 	. = ..()
-	pull_vines()
+	AddElement(/datum/element/lifesteal, 5)
+	var/datum/action/cooldown/mob_cooldown/vine_tangle/tangle = new(src)
+	tangle.Grant(src)
+	ai_controller.set_blackboard_key(BB_HUMANTRAP_VINE_TANGLE, tangle)
 
-/mob/living/simple_animal/hostile/venus_human_trap/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+/mob/living/basic/venus_human_trap/Life(seconds_per_tick = SSMOBS_DT, times_fired)
 	. = ..()
-	pixel_x = base_pixel_x + (dir & (NORTH|WEST) ? 2 : -2)
+	if(!..())
+		return FALSE
+	var/list/kudzu_in_area = range(2, src)
 
-/mob/living/simple_animal/hostile/venus_human_trap/AttackingTarget()
-	. = ..()
-	if(isliving(target))
-		var/mob/living/L = target
-		if(L.stat != DEAD)
-			adjustHealth(-maxHealth * 0.1)
-
-/mob/living/simple_animal/hostile/venus_human_trap/OpenFire(atom/the_target)
-	for(var/datum/beam/B in vines)
-		if(B.target == the_target)
-			pull_vines()
-			ranged_cooldown = world.time + (ranged_cooldown_time * 0.5)
-			return
-	if(get_dist(src,the_target) > vine_grab_distance || vines.len >= max_vines)
+	if(!(locate(/obj/structure/spacevine) in kudzu_in_area))
+		apply_damage(10, BRUTE, BODY_ZONE_CHEST) //tick 5 times to die
+		balloon_alert(src, "do not leave vines!")
```

Comment:
This is silly, make this a status effect with a screen alert you can inspect.

Reply:
what

Reply:
anyway ive observed this balloon alert to pop up like every two to four seconds outside kudzu so i think its ok

Reply:
This is a required review. Don't pop up a balloon alert every two to four seconds. Make this into a status alert or some other form of graphical feedback.